checker/check_native_imports.py

```diff
@@ -0,0 +1,601 @@
+                    continue
+
+                module_path = name
+                for ext in native_extensions:
+                    if ext in module_path:
+                        module_path = module_path.split(ext)[0]
+                        break
+
+                parts = module_path.split(".")
+                clean_parts = []
+                for part in parts:
+                    if part.startswith(("cpython-", "abi3", "pypy", "cp3")):
+                        break
+                    clean_parts.append(part)
+                module_path = ".".join(clean_parts)
+
+                module_name = module_path.replace("/", ".").rstrip(".")
+
+                if module_name:
+                    modules.append(module_name)
+
+        if not modules and has_native_files and top_level_packages:
+            return top_level_packages, top_level_packages
+
+        return sorted(set(modules)), top_level_packages
+
+    except (zipfile.BadZipFile, OSError, IOError) as e:
+        print(f"Warning: Cannot read wheel file {wheel_path}: {e}")
+        return [], []
+
+
+def test_wheel_with_logging(
+    wheel_path: Path,
+    packages_dir: Path,
+    python_bin: Path,
+    whitelist: Dict[str, Dict[str, str]],
+    used_whitelist_lock: threading.Lock,
+) -> PackageTestResult:
+    """Test wheel by installing and importing native modules with logging."""
+    wheel_name = wheel_path.name
+    result = PackageTestResult(wheel_name)
+    used_whitelist_local: Set[Tuple[str, str]] = set()
+
+    result.add_log(f"\nTesting: {wheel_name}")
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_path = Path(temp_dir)
+        venv_path = temp_path / ".venv"
+
+        try:
+            proc_result = subprocess.run(
+                [str(python_bin), "-m", "venv", str(venv_path)],
+                capture_output=True,
+                text=True,
+                encoding="utf-8",
+                timeout=180,
+            )
+        except subprocess.TimeoutExpired:
+            result.add_log("  ❌ Timeout creating venv")
+            result.finish(False, [("venv", "Timeout creating virtual environment")], [])
+            return result
+
+        if proc_result.returncode != 0:
+            result.add_log("  ❌ Failed to create venv")
+            error_msg = (
+                proc_result.stderr.strip()
+                if proc_result.stderr
+                else "Failed to create virtual environment"
+            )
+            result.finish(False, [("venv", error_msg)], [])
+            return result
+
+        result.add_log("  ✓ Created venv")
+
+        venv_python = get_venv_python(venv_path)
+
+        if not venv_python.exists():
+            result.add_log(f"  ❌ venv Python not found at {venv_python}")
+            result.finish(
+                False, [("venv", f"Python executable not found at {venv_python}")], []
+            )
+            return result
+
+        try:
+            proc_result = subprocess.run(
```

Comment:
I'm afraid this may lead to unexpected installed packages location, if executed without
```bash
VIRTUAL_ENV
PYTHONHOME
PYTHONPATH
```
environment variables. `venv/bin/activate` sets this variables and venv is expected to be used with them.
I just do not know if python would correctly place modules when executed without them.